Open Judge System/OJS.Services/Business/Contests/ContestsBusinessService.cs

```diff
@@ -15,5 +16,34 @@ public class ContestsBusinessService : IContestsBusinessService
             this.contestsData
                 .GetByIdQuery(contestId)
                 .Any(c => !c.AllowedIps.Any() || c.AllowedIps.Any(ai => ai.Ip.Value == ip));
+
+        public bool CanUserCompeteByContestUserAndIsAdminOrLecturer(Contest contest, string userId, bool isAdminOrLecturer)
```

Comment:
not good idea to have so complex object as parameter. Better pass `contestId` as parameter IMO